cli/template_render.go

```diff
@@ -0,0 +1,139 @@
+)
+
+var defaultData = template.Data{
+	Receiver: "receiver",
+	Status:   "alertstatus",
+	Alerts: template.Alerts{
+		template.Alert{
+			Status: "alertstatus",
+			Labels: template.KV{
+				"label1":          "value1",
+				"label2":          "value2",
+				"instance":        "foo.bar:1234",
+				"commonlabelkey1": "commonlabelvalue1",
+				"commonlabelkey2": "commonlabelvalue2",
+			},
+			Annotations: template.KV{
+				"annotation1":          "value1",
+				"annotation2":          "value2",
+				"commonannotationkey1": "commonannotationvalue1",
+				"commonannotationkey2": "commonannotationvalue2",
+			},
+			StartsAt:     time.Now().Add(-5 * time.Minute),
+			EndsAt:       time.Now(),
+			GeneratorURL: "https://generatorurl.com",
+			Fingerprint:  "fingerprint1",
+		},
+		template.Alert{
+			Status: "alertstatus",
+			Labels: template.KV{
+				"foo":             "bar",
+				"baz":             "qux",
+				"commonlabelkey1": "commonlabelvalue1",
+				"commonlabelkey2": "commonlabelvalue2",
+			},
+			Annotations: template.KV{
+				"aaa":                  "bbb",
+				"ccc":                  "ddd",
+				"commonannotationkey1": "commonannotationvalue1",
+				"commonannotationkey2": "commonannotationvalue2",
+			},
+			StartsAt:     time.Now().Add(-10 * time.Minute),
+			EndsAt:       time.Now(),
+			GeneratorURL: "https://generatorurl.com",
+			Fingerprint:  "fingerprint2",
+		},
+	},
+	GroupLabels: template.KV{
+		"grouplabelkey1": "grouplabelvalue1",
+		"grouplabelkey2": "grouplabelvalue2",
+	},
+	CommonLabels: template.KV{
+		"commonlabelkey1": "commonlabelvalue1",
+		"commonlabelkey2": "commonlabelvalue2",
+	},
+	CommonAnnotations: template.KV{
+		"commonannotationkey1": "commonannotationvalue1",
+		"commonannotationkey2": "commonannotationvalue2",
+	},
+	ExternalURL: "https://example.com",
+}
+
+type templateRenderCmd struct {
+	templateFilesGlobs []string
+	templateType       string
+	templateText       string
+	templateFile       *os.File
+}
+
+func configureTemplateRenderCmd(cc *kingpin.CmdClause) {
+	var (
+		c         = &templateRenderCmd{}
+		renderCmd = cc.Command("render", "Render a given definition in a template file to standard output")
+	)
+
+	renderCmd.Flag("templateglob", "Glob of paths that will be expanded and used for rendering").Required().StringsVar(&c.templateFilesGlobs)
+	renderCmd.Flag("templatetext", "The template that will be rendered").Required().StringVar(&c.templateText)
+	renderCmd.Flag("templatetype", "The type of the template. Can be either text (default) or html").EnumVar(&c.templateType, "html", "text")
+	renderCmd.Flag("templatefile", `Full path to a file which contains the data of the alert(-s) with which the --templatetext will be rendered. Must be in JSON. File must be formatted according to the following layout: https://pkg.go.dev/github.com/prometheus/alertmanager/template#Data. If none has been specified then a predefined, simple alert will be used for rendering`).FileVar(&c.templateFile)
```

Comment:
I think this name makes it sounds like it's maybe an alternative to `templateglob` (i.e. use this specific file).
How about `templatedata` or `datasource`?

Reply:
I didn't think about this similarity. `templatedata` sounds good, will rename it. BTW @mrliptontea have you ever seen a similar tool online? I haven't and I'm surprised that no one has made this yet

Reply:
Found https://github.com/prometheus/alertmanager/pull/2414/files :disappointed:

Reply:
promslack was the closest to what I needed but sadly it's not quite it.
I saw that other Pull Request but I prefer this - it's simpler yet fulfils my needs.

I'm now writing some "tests" where I have a bunch of different alerts in JSON, I render the templates, and compare the output to make the alerts more user friendly so thanks @GiedriusS you made it possible :slightly_smiling_face:

Reply:
```suggestion
	renderCmd.Flag("template.file", `Full path to a file which contains the data of the alert(-s) with which the --template.text will be rendered. Must be in JSON. File must be formatted according to the following layout: https://pkg.go.dev/github.com/prometheus/alertmanager/template#Data. If none has been specified then a predefined, simple alert will be used for rendering`).FileVar(&c.templateFile)
```